File: 201804/calc.py

def quarters(qRemaining, nDay):
    if nDay==0: return qRemaining
    if qRemaining % 6: return None
    qPrior = nDay + 7*qRemaining/6
    return quarters(qPrior, nDay-1)

solutions = filter(lambda x:x[1] != None,
                   [(nDays,quarters(0,nDays))
                    for nDays in range(4,10**7)])

print ' '.join(map(str,solutions))

File: 201804/calc.hs
quarters:: Int -> Int -> Maybe Int
quarters qRemaining 0 = Just qRemaining
quarters qRemaining nDay
    | (qRemaining `mod` 6) > 0  = Nothing
    | otherwise = quarters qPrior (nDay-1)
                  where qPrior = nDay + 7*qRemaining `quot` 6

solutions = [(nDays, qs) | nDays <-[4..10^7],
                           let qs = quarters 0 nDays,
                           qs /= Nothing]

main = putStrLn.unwords $ map show solutions
